### PR TITLE
Use naming conventions and comments to explain what timeouts really mean

### PR DESCRIPTION
**Pull request by lukebakken**

### CODE REVIEW

File: src/riakc_pb_socket.erl [hunk 1]
```diff
@@ -534,112 +563,123 @@ stream_list_keys(Pid, Bucket) ->
...
+    CT = default_timeout(get_bucket_timeout),
+    get_bucket(Pid, Bucket, CT).
 
 %% @doc Get bucket properties specifying a server side timeout.
 %% @equiv get_bucket(Pid, Bucket, Timeout, default_timeout(get_bucket_call_timeout))
 -spec get_bucket(pid(), bucket() | bucket_and_type(), timeout()) -> {ok, bucket_props()} | {error, term()}.
-get_bucket(Pid, Bucket, Timeout) ->
-    get_bucket(Pid, Bucket, Timeout, default_timeout(get_bucket_call_timeout)).
+get_bucket(Pid, Bucket, Timeout) when is_integer(Timeout); Timeout =:= infinity ->
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Can we mark this as deprecated then?

File: src/riakc_pb_socket.erl [hunk 2]
```diff
@@ -534,112 +563,123 @@ stream_list_keys(Pid, Bucket) ->
...
+    set_bucket(Pid, Bucket, BucketProps, CT).
 
 %% @doc Set bucket properties specifying a server side timeout.
 %% @equiv set_bucket(Pid, Bucket, BucketProps, Timeout, default_timeout(set_bucket_call_timeout))
 -spec set_bucket(pid(), bucket() | bucket_and_type(), bucket_props(), timeout()) -> ok | {error, term()}.
-set_bucket(Pid, Bucket, BucketProps, Timeout) ->
-    set_bucket(Pid, Bucket, BucketProps, Timeout,
-               default_timeout(set_bucket_call_timeout)).
+set_bucket(Pid, Bucket, BucketProps, Timeout) when is_integer(Timeout); Timeout =:= infinity ->
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate?

File: src/riakc_pb_socket.erl [hunk 3]
```diff
@@ -534,112 +563,123 @@ stream_list_keys(Pid, Bucket) ->
...
+    CT = default_timeout(set_bucket_call_timeout),
+    set_bucket(Pid, Bucket, BucketProps, Timeout, CT).
 
 %% @doc Set bucket properties specifying a server side and local call timeout.
 %% @deprecated because `CallTimeout' is ignored
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: :awwyiss: deprecate like this.

File: src/riakc_pb_socket.erl [hunk 4]
```diff
@@ -534,112 +563,123 @@ stream_list_keys(Pid, Bucket) ->
...
+    call_infinity(Pid, {req, Rpb, Timeout}).
 
 %% @doc Reset bucket properties back to the defaults.
 %% @equiv reset_bucket(Pid, Bucket, default_timeout(reset_bucket_timeout), default_timeout(reset_bucket_call_timeout))
 -spec reset_bucket(pid(), bucket() | bucket_and_type()) -> ok | {error, term()}.
 reset_bucket(Pid, Bucket) ->
-    reset_bucket(Pid, Bucket, default_timeout(reset_bucket_timeout), default_timeout(reset_bucket_call_timeout)).
+    ST = default_timeout(reset_bucket_timeout),
+    CT = default_timeout(reset_bucket_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate/remove.

File: src/riakc_pb_socket.erl [hunk 5]
```diff
@@ -534,112 +563,123 @@ stream_list_keys(Pid, Bucket) ->
...
+    reset_bucket(Pid, Bucket, ST, CT).
 
 %% @doc Reset bucket properties back to the defaults.
 %% @equiv reset_bucket(Pid, Bucket, Timeout, default_timeout(reset_bucket_call_timeout))
 -spec reset_bucket(pid(), bucket() | bucket_and_type(), timeout()) -> ok | {error, term()}.
-reset_bucket(Pid, Bucket, Timeout) ->
-    reset_bucket(Pid, Bucket, Timeout, default_timeout(reset_bucket_call_timeout)).
+reset_bucket(Pid, Bucket, Timeout) when is_integer(Timeout); Timeout =:= infinity ->
+    CT = default_timeout(reset_bucket_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate

File: src/riakc_pb_socket.erl [hunk 6]
```diff
@@ -662,7 +703,9 @@ mapred(Pid, Inputs, Query, Timeout) ->
                     {error, {badqterm, mapred_queryterm()}} |
                     {error, timeout} |
                     {error, term()}.
-mapred(Pid, Inputs, Query, Timeout, CallTimeout) ->
+mapred(Pid, Inputs, Query, Timeout, CallTimeout)
+  when is_integer(Timeout); Timeout =:= infinity ->
+    % TODO CallTimeout is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate.

File: src/riakc_pb_socket.erl [hunk 7]
```diff
@@ -692,14 +739,20 @@ mapred_stream(Pid, Inputs, Query, ClientPid) ->
...
                            {error, {badqterm, mapred_queryterm()}} |
                            {error, timeout} |
                            {error, Err :: term()}.
-mapred_stream(Pid, Inputs, Query, ClientPid, Timeout) ->
-    mapred_stream(Pid, Inputs, Query, ClientPid, Timeout,
-                  default_timeout(mapred_stream_call_timeout)).
+mapred_stream(Pid, Inputs, Query, ClientPid, Timeout)
+  when is_integer(Timeout); Timeout =:= infinity ->
+    CT = default_timeout(mapred_stream_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate.

File: src/riakc_pb_socket.erl [hunk 8]
```diff
@@ -759,7 +817,9 @@ mapred_bucket(Pid, Bucket, Query) ->
                            {error, timeout} |
                            {error, Err :: term()}.
 mapred_bucket(Pid, Bucket, Query, Timeout) ->
-    mapred_bucket(Pid, Bucket, Query, Timeout, default_timeout(mapred_bucket_call_timeout)).
+    CT = default_timeout(mapred_bucket_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate / remove.

File: src/riakc_pb_socket.erl [hunk 9]
```diff
@@ -790,9 +851,11 @@ mapred_bucket(Pid, Bucket, Query, Timeout, CallTimeout) ->
 -spec mapred_bucket_stream(ConnectionPid::pid(), bucket(), [mapred_queryterm()], ClientPid::pid(), timeout()) ->
                                   {ok, req_id()} |
                                   {error, term()}.
-mapred_bucket_stream(Pid, Bucket, Query, ClientPid, Timeout) ->
-    mapred_bucket_stream(Pid, Bucket, Query, ClientPid, Timeout,
-                         default_timeout(mapred_bucket_stream_call_timeout)).
+mapred_bucket_stream(Pid, Bucket, Query, ClientPid, Timeout)
+  when is_integer(Timeout); Timeout =:= infinity ->
+    CT = default_timeout(mapred_bucket_stream_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate.

File: src/riakc_pb_socket.erl [hunk 10]
```diff
@@ -823,32 +886,37 @@ search(Pid, Index, SearchQuery) ->
...
 %%      unless executed against a Riak Search cluster.
 -spec search(pid(), binary(), binary(), search_options(), timeout()) ->
                     {ok, search_result()} | {error, term()}.
-search(Pid, Index, SearchQuery, Options, Timeout) ->
-    CallTimeout = default_timeout(search_call_timeout),
-    search(Pid, Index, SearchQuery, Options, Timeout, CallTimeout).
+search(Pid, Index, SearchQuery, Options, Timeout)
+  when is_integer(Timeout); Timeout =:= infinity ->
+    CT = default_timeout(search_call_timeout),
+    % TODO CT is ignored and should be removed
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Deprecate.

File: src/riakc_pb_socket.erl [hunk 11]
```diff
@@ -1170,10 +1227,29 @@ default_timeout(OpTimeout) ->
...
 
+-spec get_timeouts(list(proplists:property())) -> {timeout(), timeout()}.
+get_timeouts(Options) ->
+    ST = case proplists:get_value(timeout, Options) of
+             undefined -> ?DEFAULT_PB_TIMEOUT;
+             T -> T
+         end,
+    CT = if
+             is_integer(ST) ->
+                 %% Add an extra 500ms to the create_search_index timeout
```
Comment (alexmoore) [inline src/riakc_pb_socket.erl]: Is this only used for search timeouts or others that stream too?

**Review by alexmoore — APPROVED** [review]
Making erlang more readable? This is crazy talk. 

+1 if we mark the appropriate functions as deprecated (see notes).

**Review by christophermancini — APPROVED** [review]
Only approving this to get it off my list of review requested PRs.

**Pull request closed without merging**